src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -319,6 +320,60 @@ public void AddBytes(ReadOnlySpan<byte> value)
             ref byte pos = ref MemoryMarshal.GetReference(value);
             ref byte end = ref Unsafe.Add(ref pos, value.Length);
 
+            if (Unsafe.ByteOffset(ref pos, ref end) < (sizeof(int) * 4))
+            {
+                goto Small;
+            }
+
+            // Usually Add calls Initialize but if we haven't used HashCode before it won't have been called.
+            if (_length == 0)
+            {
+                Initialize(out _v1, out _v2, out _v3, out _v4);
+            }
+            else
+            {
+                // If we have more than 16 bytes to hash, we can add them in 16-byte batches,
+                // but we first have to add enough data to flush any queued values.
+                switch (_length % 4)
```

Comment:
Do we have appropriate test coverage for calling AddBytes after 0, 1, 2, and 3 other ints have been added, and with < 16, 16, and > 16 bytes?

Reply:
[The existing tests around `AddBytes`](https://github.com/dotnet/runtime/blob/5877e8b713742b6d80bd1aa9819094be029e3e1f/src/libraries/System.Runtime/tests/System/HashCodeTests.cs#L126-L188) seem adequate to me. They have already caught one mistake of mine, and the method itself has a generous amount of asserts.

Reply:
I have verified that the existing tests achieve 100% branch coverage on `AddBytes` and `Add(int)`.